src/zarr/store/zip.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import annotations
+
+import os
+import threading
+import time
+import zipfile
+from collections.abc import AsyncGenerator
+from pathlib import Path
+from typing import Literal
+
+from zarr.abc.store import Store
+from zarr.buffer import Buffer, BufferPrototype
+
+ZipStoreAccessModeLiteral = Literal["r", "w", "a"]
+
+
+class ZipStore(Store):
+    supports_writes: bool = True
+    supports_partial_writes: bool = False
+    supports_listing: bool = True
```

Comment:
what do we think about adding `supports_deletes: bool = False` as a class attribute?

Reply:
that would be a solid extrapolation of the design we are currently using. one downside of that design is that, even if a class has `supports_x` set to `False`, the class will still need an implementation of `x`. Another solution would be to express `supports_x` by having the class inherit from a `DoesX` mixin. But that's maybe out of scope for this PR.

Reply:
I've got the `supports_deletes` route for now.